runner/components/io.siddhi.distribution.metrics.core/src/main/java/io/siddhi/distribution/metrics/prometheus/reporter/impl/PrometheusReporter.java

```diff
@@ -0,0 +1,158 @@
+        super(name);
+        this.metricRegistry = metricRegistry;
+        this.metricFilter = metricFilter;
+        this.serverURL = serverURL;
+    }
+
+    @Override
+    public void startReporter() {
+
+        prometheusReporter = PrometheusReporter.forRegistry(metricRegistry, serverURL).filter(metricFilter)
+                .convertRatesTo(TimeUnit.SECONDS).convertDurationsTo(TimeUnit.MILLISECONDS).build();
+
+        try {
+            URL target = new URL(serverURL);
+            collectorRegistry = new CollectorRegistry();
+            collectorRegistry.register(new DropwizardExports(metricRegistry));
+
+            initiateServer(target.getHost(), target.getPort());
+            log.info("Prometheus Server has successfully connected at " + serverURL);
+
+        } catch (IOException | ConnectionUnavailableException e) {
+            log.error("Prometheus Server connection failed at " + serverURL, e);
+        }
+    }
+
+    @Override
+    public void stopReporter() {
+        if (prometheusReporter != null) {
+            disconnect();
+            destroy();
+            prometheusReporter.stop();
+            prometheusReporter = null;
+
+        }
+
+    }
+
+    public static PrometheusReporter.Builder forRegistry(MetricRegistry registry, String serverURL) {
+        return new PrometheusReporter.Builder(registry, serverURL);
+    }
+
+    private void initiateServer(String host, int port) throws ConnectionUnavailableException {
+        try {
+            InetSocketAddress address = new InetSocketAddress(host, port);
+            server = new HTTPServer(address, collectorRegistry);
+        } catch (IOException e) {
+            if (!(e instanceof BindException && e.getMessage().equals("Address already in use"))) {
+                log.error("Unable to establish connection for Prometheus  \'\' at " + serverURL, e);
+                throw new ConnectionUnavailableException("Unable to establish connection for Prometheus \'\' at "
+                        + serverURL, e);
+            }
+        }
+    }
+
+    private void disconnect() {
+        if (server != null) {
+            server.stop();
+            log.info("Prometheus Server successfully stopped at " + serverURL);
+        }
+    }
+
+    private void destroy() {
+        if (collectorRegistry != null) {
+            collectorRegistry.clear();
+        }
+    }
+
+    /**
+     * Builds a {@link PrometheusReporter} with the given properties.
+     *
+     * @return a {@link PrometheusReporter}
+     */
+    public static class Builder {
+        private final MetricRegistry registry;
+        private MetricFilter filter;
+        private String serverURL;
+
+        private Builder(MetricRegistry registry, String serverURL) {
+            this.registry = registry;
+            this.serverURL = serverURL;
+            this.filter = MetricFilter.ALL;
+        }
+
+        public Builder setServerURL(String serverURL) {
```

Comment:
Is this used?

Reply:
The above comment is addressed by b0e138f

Reply:
No, its seems to be an unused method. any reason to add it?

Reply:
The above comment is addressed by the commit 9a0bfe0.